app/src/main/java/com/atomicrobot/carbon/ui/compose/MainNavigation.kt

```diff
@@ -0,0 +1,42 @@
+package com.atomicrobot.carbon.ui.compose
+
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import com.atomicrobot.carbon.StartActivity
+import com.atomicrobot.carbon.ui.deeplink.DeepLinkSampleScreen
+import com.atomicrobot.carbon.ui.main.Main
+import com.atomicrobot.carbon.ui.main.MainViewModel
+import com.atomicrobot.carbon.ui.splash.SplashScreen
+import com.atomicrobot.carbon.ui.splash.SplashViewModel
+
+@Composable
+fun MainNavigation(isDeepLinkIntent: Boolean) {
+    val viewModel = hiltViewModel<SplashViewModel>()
+    val mainViewModel = hiltViewModel<MainViewModel>()
+    val navController = rememberNavController()
+
+    NavHost(navController = navController, startDestination = StartActivity.splashPage, modifier = Modifier) {
+        composable(StartActivity.splashPage) {
+            SplashScreen(){
+                navController.popBackStack()
+                if (isDeepLinkIntent) {
+                    navController.navigate(viewModel.getDeepLinkNavDestination())
+                }
+                else {
+                    navController.navigate(StartActivity.mainPage)
+                }
+            }
+
+        }
+        composable(StartActivity.mainPage) { Main(mainViewModel) }
```

Comment:
Do we want the main screen to just be called `Main()`? On the koin branch I called it `MainScreen` to match with `SplashScreen` etc.

Reply:
Adding "Screen" sounds like a good way to go